apps/federatedfilesharing/lib/Migration/Version1011Date20201120125158.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/*
+ * @copyright Copyright (c) 2020 Julius Härtl <jus@bitgrid.net>
+ *
+ * @author Julius Härtl <jus@bitgrid.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+declare(strict_types=1);
+
+namespace OCA\FederatedFileSharing\Migration;
+
+use Closure;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\DBAL\Types\Types;
+use OCP\DB\ISchemaWrapper;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+class Version1011Date20201120125158 extends SimpleMigrationStep {
+
+	/** @var IDBConnection */
+	private $connection;
+
+	public function __construct(IDBConnection $connection) {
+		$this->connection = $connection;
+	}
+
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options) {
+		/** @var ISchemaWrapper $schema */
+		$schema = $schemaClosure();
+
+		if ($schema->hasTable('federated_reshares')) {
+			$table = $schema->getTable('federated_reshares');
+			$remoteIdColumn = $table->getColumn('remote_id');
+			if ($remoteIdColumn && $remoteIdColumn->getType()->getName() !== Types::STRING) {
+				$remoteIdColumn->setNotnull(false);
+				$remoteIdColumn->setType(Type::getType(Types::STRING));
+				$remoteIdColumn->setOptions(['length' => 255]);
+				$remoteIdColumn->setDefault('');
+			}
+		}
+
+		return $schema;
```

Comment:
```suggestion
				$remoteIdColumn->setDefault('');
				return $schema;
			}
		}

		return null;
```

save some seconds in case of rerunning

Reply:
Done